### DISCUSSION TITLE
2FA

### DISCUSSION THREAD

**Discussion by ghost** [community/community#69829] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

General

### Body

I am not about to use a fucking smartphone to log into my GitHub account to have full functionality. I fucking abhor that branch of technology. Nor am I about to go through the SMS way of setting 2FA up, **I'M NOT GIVING YOU MY PHONE NUMBER BITCH!!**

Either I find some janky way of bypassing or setting it up without any of that nonsense or I am going to delete everything on my account and then the account itself. I really do not need this shit right now, go fuck yourself GitHub!

**Comment by FogoVoar** [comment]
Even if I wanted to surrender to 2FA (which I don't want) I couldn't. None of the third-party app options work on my outdated devices, nor is the SMS option available in my country. For me this is the biggest problem, they are ignoring everyone who not only doesn't want to join this system, but who is unable to do so. This is a big step backwards in the software industry, painful to watch.

**Reply by ghostinhershell** [reply]
Hey there @FogoVoar, 

Thanks so much for sharing this feedback in our community. I want to focus on the feedback that you've shared here:
> None of the third-party app options work on my outdated devices, nor is the SMS option available in my country.

It's important because it does demonstrate that there may be edge cases where our current multi factor authentication options may not be accessible for all users. I've filed feedback with our Accessibility team so that they can take a closer look. Can you share a bit more information here, like what country you are based in and the device that you currently have that doesn't support TOTP apps?

**Reply by ghostinhershell** [reply]
And actually, do you mind creating a separate discussion post for your particular issue? Feel free to file it in the Accessibility community using this form here: https://github.com/orgs/community/discussions/new?category=accessibility.

**Reply by kheitmannf** [reply]
I also have the same problem. I don't trust google TOTP because I did tested my recovery codes in 3 services that I had to enable 2FA, and the recovery codes didn't work. Now Github is telling me to configure 2FA and the ONLY option they give me is TOTP (my country is not in the SMS list). Why can't you understand it? If I lose my phone and I depend on TOTP I'll get locked out of my account definitively. But if I lose my phone and I depend on SMS, I can talk with the company, recover my phone number, and use my account again. TOTP is more comfortable, is quicker, etc. But in the unlikely scenario it fails and recovery keys also fail.... I have much more to lose compared to using SMS authentication. SMS is much more robust. Why you don't fill in the list with the countries that are missing? Because you want EVERYBODY to use TOTP, and you BELIEVE that TOTP is the best way to do authentication. That's bully and tyrannic. Let people use the authentication method people want to use. Provide all the options to your customers and let the customers take the decision.

**Comment by kuroodo** [comment]
Tomorrow I will lose access to my github account forever because of this. I recommend you move to GitLab. They at least have 2FA through email which is just as secure if not better for the average user.

**Comment by AwokeKnowing** [comment]
I recommend everyone leave github

They are now just become bullies and another Microsoft app. 

They are unfit to host the open source community's work. Now they don't even let you decide how to authenticate. they force you to tie your account to SMS or some app. 

Move on.  

Github has lost it's way.Forcing 2fa against dev's will is EXACTLY the kind of behavior that made MS infamous. Why kill github? why don't you get it? MAKE 2FA OPTIONAL

**Comment by ghost** [comment]
Well I just got asked again to set up this 2FA shit. Now that I know this isn't just going to be a one time thing I can throw a fake phone number at and will occasionally be nagged to go through it again I'm just going to go ahead and fucking delete everything. I wish to see you bathe in eternal fire MS and GitHub, FUCK YOU!
